x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/PhysicalVerifier.java

```diff
@@ -14,10 +14,13 @@
 import org.elasticsearch.xpack.esql.optimizer.rules.PlanConsistencyChecker;
 import org.elasticsearch.xpack.esql.plan.logical.Enrich;
 import org.elasticsearch.xpack.esql.plan.physical.EnrichExec;
+import org.elasticsearch.xpack.esql.plan.physical.EsQueryExec;
 import org.elasticsearch.xpack.esql.plan.physical.FieldExtractExec;
 import org.elasticsearch.xpack.esql.plan.physical.PhysicalPlan;
 
+import static org.elasticsearch.index.IndexMode.LOOKUP;
 import static org.elasticsearch.xpack.esql.common.Failure.fail;
+import static org.elasticsearch.xpack.esql.core.expression.Attribute.datatypeEquals;
 
 /** Physical plan verifier. */
 public final class PhysicalVerifier {
```

Comment:
Maybe out of scope: the physical and logical verifiers share quite a bit of code and behavior now; maybe we could have them inherit from the same (abstract) base class by using that `PhysicalPlan` and `LogicalPlan` both inherit from `QueryPlan`. By skimming this, I didn't notice anything that super duper depends on methods of `LogicalPlan` or `PhysicalPlan`.